app/models/operation.rb

```diff
@@ -50,9 +50,13 @@ def debit!(reference:, amount:, kind: :main, currency: nil)
       )
     end
 
-    def balance(currency: nil)
+    def balance(currency: nil, date_range: nil)
```

Comment:
Parse date_range in controller and pass ActiveSupport::TimeWithZone to Operatios #balance

```suggestion
    def balance(currency: nil, created_at_from: nil, created_at_to: nil)
      db_balances.where(created_at: created_at_from..created_at_to)
```